projects/packages/videopress/src/client/block-editor/blocks/video/components/videopress-player/index.native.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * External dependencies
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { SandBox } from '@wordpress/components';
+import { useCallback, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { View, Text } from 'react-native';
+
+/**
+ * VideoPlayer react component
+ *
+ * @param {object} props                 - Component props.
+ * @param {object} props.html            - Player html to render in the sandbox.
+ * @param {object} props.attributes      - Block attributes.
+ * @param {Array} props.scripts          - Scripts to pass trough to the sandbox.
+ * @param {object} props.preview         - oEmbed preview data.
+ * @param {boolean} props.isRequestingEmbedPreview - oEmbed preview data.
+ * @returns {object}                     - React component.
+ */
+export default function VideoPressPlayer( {
+	html,
+	attributes,
+	scripts = [],
+	preview,
+	isRequestingEmbedPreview,
+} ) {
+	const [ videoPlayerHeight, setVideoPlayerHeight ] = useState( 250 );
+
+	// Use preview aspect ratio if available. Otherwise, use the ratio from the attributes.
+	const { height = 0, width = 1 } = preview;
+	const videoRatio = ( height / width ) * 100 || attributes.videoRatio;
+
+	// If we don't have `html` then the player is loading.
+	const loadingVideoPlayer = ! html;
+
+	const embedPreviewRequested = ! isRequestingEmbedPreview;
+
+	// Scale video player height based on the video ratio and layout width.
+	const scaleVideoPlayer = useCallback(
+		( { layout } ) => !! videoRatio && setVideoPlayerHeight( ( layout.width * videoRatio ) / 100 ),
+		[ videoRatio ]
+	);
+
+	return (
+		<View
+			style={ { height: videoPlayerHeight } }
+			onLayout={ ( { nativeEvent } ) => scaleVideoPlayer( nativeEvent ) }
+		>
+			{ embedPreviewRequested && <SandBox html={ html } scripts={ scripts } /> }
```

Comment:
It's likely that we'd need to pass a custom JS script to communicate with the player, especially in relation to fetching the token for private videos. As an example, [here](https://github.com/Automattic/wp-calypso/blob/9b66da6ae2297cd78b19cad5982e66a516b2ca94/client/post-editor/media-modal/detail/detail-preview-videopress.jsx#L113-L146) is how Calypso handles the token request when asked from the player.

Reply:
Agreed, however that feels like a conversation for future changes.
The scope here is to get a native version of the player working for public videos.

Reply:
Good point. I agree that if we limit the scope to public videos this won't be necessary 👍. Let's keep this in mind in the next iteration but not as a blocker of the PR.